docs/source/format/Columnar.rst

```diff
@@ -1385,6 +1385,36 @@ have two entries in each RecordBatch. For a RecordBatch of this schema with
     buffer 13: col2    data
 
 
+Compression
+-----------
+
+There are three different options for record batch body
+buffers compression: buffers can be uncompressed, can use
+``lz4`` or ``zstd`` compression codec. All buffers in the flat
+sequence of the message body are compressed separately with the
+same codec.
```

Comment:
Small suggestion to improve clarity. This makes it more verbose, but I think it's worthwhile:
```suggestion
There are three different options for compression of record batch
body buffers: Buffers can be uncompressed, buffers can be
compressed with the``lz4`` compression codec, or buffers can
be compressed with the ``zstd`` compression codec. Buffers in
the flat sequence of a message body must be either all
uncompressed or all compressed separately using the same codec.
```